src/macro.jl

```diff
@@ -90,17 +102,23 @@ macro irrational(sym, val, def, T=Symbol(uppercasefirst(string(sym))))
         end
     else
         # newer Julia versions
-        isa(def, Symbol) ? quote
-            function Base.BigFloat(::$eT, r::Base.MPFR.MPFRRoundingMode=Base.MPFR.ROUNDING_MODE[]; precision=precision(BigFloat))
-                c = BigFloat(; precision=precision)
-                ccall(($(string("mpfr_const_", def)), :libmpfr),
-                      Cint, (Ref{BigFloat}, Base.MPFR.MPFRRoundingMode), c, r)
-                return c
+        if isa(def, Symbol)
+            # support older Julia versions prior to https://github.com/JuliaLang/julia/pull/51362
+            r = VERSION < v"1.12.0-DEV.78" ? :(Base.MPFR.ROUNDING_MODE[]) : :(Base.Rounding.rounding_raw(BigFloat))
+            quote
+                function Base.BigFloat(::$eT, r::Base.MPFR.MPFRRoundingMode=$r; precision=precision(BigFloat))
```

Comment:
`rounding_raw` exists way back to v1.0

```suggestion
            quote
                function Base.BigFloat(::$eT, r::Base.MPFR.MPFRRoundingMode=Base.Rounding.rounding_raw(BigFloat); precision=precision(BigFloat))
```

Reply:
As far as I understood, `Base.@irrational` correctly used `MPFR.ROUNDING_MODE[]` as default value until https://github.com/JuliaLang/julia/pull/51362. In that PR it should have been changed to `Rounding.rounding_raw(BigFloat)` which was fixed in a later PR. Since generally IrrationalConstants intends to mainly copy the implementation of `Base.@irrational`, intentionally I only switched to `Rounding.rounding_raw(BigFloat)` from https://github.com/JuliaLang/julia/pull/51362 onwards.

Reply:
I understand what you did. I'm saying it's unnecessary.

Reply:
One way to look at it is that doing `MPFR.ROUNDING_MODE[]` was only correct by accident, because the implementation of `rounding_raw` was this:

https://github.com/JuliaLang/julia/blob/501a4f25c2b7626c5e368cd01d1c050b70bafdb9/base/mpfr.jl#L161-L161

Reply:
But there's absolutely no reason to inline the implementation of `rounding_raw` on purpose. That was a bug.

Reply:
Furthermore, accessing either `rounding_raw` or `ROUNDING_MODE` is trespassing on `Base` internals, however `ROUNDING_MODE` is, as the implementation detail of an implementation detail, clearly the worse option.

Reply:
Furthermore, doing things conditionally on the version of Julia isn't exactly a good practice.

Reply:
This package intentionally reimplements and slightly modifies Base internals, so there's no way around accessing internals and dealing with different Julia versions.

> there's absolutely no reason to inline the implementation of `rounding_raw` on purpose.

In Base you mean? Not saying it was a good idea, but `ROUNDING_MODE` was used all over the place in Base as default value prior to https://github.com/JuliaLang/julia/pull/51362 as the diff shows.

Generally, for consistency reasons for each Julia version I'd rather stick with the implementation of `Base.@irrational`.

Reply:
> In Base you mean?

Both here and in `Base`.

Reply:
> there's no way around accessing internals and dealing with different Julia versions.

Disagree. This package doesn't in essence require any of that.

Reply:
Changing https://github.com/JuliaMath/IrrationalConstants.jl/blob/872ace5826bb3f457df1545c4c69d6312c9c935e/src/macro.jl#L24 would either introduce bugs in old Julia versions or adding unnecessary redundant definitions in newer versions.

To conclude, it's a design decision of this package to use Base internals and copy the implementation and tests from Base. This is not something to be changed in this PR.